src/Groups/matrices/iso_nf_fq.jl

```diff
@@ -0,0 +1,172 @@
+# Detinko, Flannery, O'Brien "Recognizing finite matrix groups over infinite
+# fields", Section 4.2
+function isomorphic_group_over_finite_field(matrices::Vector{T}) where T <: Union{ fmpq_mat, AbstractAlgebra.MatElem{nf_elem} }
+   @assert !isempty(matrices)
+
+   # One should probably check whether all matrices are n by n (and invertible
+   # and such ...)
+
+   K = base_ring(matrices[1])
+   n = nrows(matrices[1])
+
+   Fq, matrices_Fq = good_reduction(matrices, 2)
+
+   G = MatrixGroup(n, Fq, matrices_Fq)
+   N = GAP.Globals.Order(G.X)
+   if maximal_order_of_finite_linear_group_qq(degree(K)*n) < N
+      error("Group is not finite")
+   end
+
+   G_to_fin_pres = GAP.Globals.IsomorphismFpGroupByGenerators(G.X, GAP.julia_to_gap([ g.X for g in gens(G) ]))
```

Comment:
can't you use `isomorphic_fp_group(G)` here?

Reply:
`isomorphic_fp_group` uses `IsomorphismFpGroup` in GAP. But I want a presentation where the relations are exactly in the generators I defined the group with. From the GAP manual I understood that `IsomorphismFpGroupByGenerators` is the right one for that.

Reply:
OK, makes sense. Then we should add a wrapper for that. I've started a new issue https://github.com/oscar-system/Oscar.jl/issues/541 with "Wishlist groups interface" ; feel free to add things there you also need